lib/utils.js

```diff
@@ -192,6 +192,8 @@ function resolveCompactionCollision(
     const otherItem = layout[i];
     // Ignore static items
     if (otherItem.static) continue;
+    // Don't resolve collision if with itself
+    if (otherItem.i === item.i) continue;
```

Comment:
This shouldn't be possible given that we iterate through the list starting past this item. Have you seen it compare with itself before?